src/cbmc/bmc.cpp

```diff
@@ -377,6 +377,12 @@ safety_checkert::resultt bmct::execute(
     const goto_functionst &goto_functions =
       goto_model.get_goto_functions();
 
+    // This provides the frontend the opportunity to do things like a
+    // symbol-table or goto-functions dump instead of actually running the
+    // checker, like show-vcc except frontend specific.
+    if(frontend_callback_after_symex && frontend_callback_after_symex())
```

Comment:
The examples seem strange: why should dumping a symbol table or goto functions require running symex? show-vcc of course does.

Reply:
Because we use symex to load the functions, running show-goto-functions before symex would result in just `__CPROVER_Start`. The comment should probably make it explicit that this is used when looking at the "symbol-table... _in conjunction with symex driven loading_"

Reply:
Because with symex-driven lazy loading in play, symex is the only way functions get loaded. Thus show-goto-functions has to come after symex in that case.

Reply:
@thk123 added comment.

Reply:
Sorry, but having to do symbolic execution in order to obtain syntactic information makes no sense. It's ok to forbid lazy loading together with show-goto-functions, but please don't produce a tool that requires a semantic analysis to produce syntactic information.

Reply:
This is a reasonable thing to want for debug purposes: if I want to know what got loaded by symex-driven loading, I would pass `--symex-driven-lazy-loading --list-goto-functions` for example. If I want the whole class file I just don't use `symex-driven-lazy-loading`, which processes show/list-goto-functions the old fashioned way.

Reply:
We already have `--lazy-methods`, under which `--show-goto-functions` will again show whatever was loaded, not all the functions in a particular class file.

Reply:
It didn't claim that was good either ;-)

What I would suggest is additional text to the help output that explains the interaction of the command-line options.

Reply:
https://github.com/diffblue/cbmc/pull/1957